### ISSUE TITLE
Entity as a parameter in a Query is not correctly processed to its identifier value

### ISSUE THREAD

**Issue by fancyweb** [doctrine/doctrine2#6073]
Hello,
I've just found a problem in the ```processParameterValue``` method of the ```AbstractQuery``` class (https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/AbstractQuery.php#L423).

If you have an entity as a parameter in a ```Query```, if the ```ClassMetadata``` of this entity has not been fetched yet, the entity won't be transformed to its identifier to be used as the parameter in the final generated sql query. This is because the ```hasMetadataFor``` method is used, and this method only check in the already fetched ```ClassMetadata```.

**Comment by Ocramius** [comment]
@fancyweb ugly, but indeed true. Possibly to be solved in 3.x, when we change the metadata approach completely.

**Comment by fancyweb** [comment]
@Ocramius Couldn't we use the method ```getMetadataFor``` method in a try catch block instead ?

**Comment by Ocramius** [comment]
> in a try catch block instead ?

Yes, that can also be done, and should be simple too, for now. Needs some test case scenarios though: 

 * non-mapped class
 * non-loaded metadata
 * loaded metadata
